### ISSUE TITLE
Task :react-native-iap:compilePlayDebugKotlin FAILED

### ISSUE THREAD

**Issue by MuhammadRafeh** [dooboolab-community/react-native-iap#2520]
> Please use the [Discussion board](https://github.com/dooboolab-community/react-native-iap/discussions) if you want to get some help. Please use issues to report bugs.

**Description**

**Expected Behavior**

**Screenshots**

**Envir
<img width="1512" alt="Screenshot 2023-08-20 at 8 47 32 PM" src="https://github.com/dooboolab-community/react-native-iap/assets/61616929/b5f304b4-f005-45af-a0e2-b1b2bfc8c4a1">
onment:**

- react-native-iap:
- react-native:
- Platforms (iOS, Android, emulator, simulator, device):

**To Reproduce**

2. Press '...'
3. Error '...' is shown

---

[Optional] **Additional Context**

**Comment by ahmedrowaihi** [comment]
YES!, I was having this issue a few minutes ago, I traced the last update of this library and I found out that the Kotlin version has changed due to the release of Google Play Billing Library 5.2 increase here 
[gradle.properties#L9](https://github.com/dooboolab-community/react-native-iap/blob/dad69b1f5a8792c144b04d4af7f7986ce945ddfc/android/gradle.properties#L9) on last version [12.10.7](https://github.com/dooboolab-community/react-native-iap/releases/tag/12.10.7)


Apply this patch to your project
use [patch-package](patch-package)

steps:
1. create folder named ```patches``` on the root of your RN project
2. under ```patches``` folder create a file ```react-native-iap+12.10.7.patch``` and pase code below
3. run command ```npx patch-package```

```patch
diff --git a/node_modules/react-native-iap/android/gradle.properties b/node_modules/react-native-iap/android/gradle.properties
index e76bb7c..afc8cdd 100644
--- a/node_modules/react-native-iap/android/gradle.properties
+++ b/node_modules/react-native-iap/android/gradle.properties
@@ -6,7 +6,7 @@ RNIap_buildToolsVersion=30.0.2
 RNIap_ndkversion=21.4.7075529
 RNIap_playServicesVersion=18.1.0
 RNIap_amazonSdkVersion=3.0.3
-RNIap_playBillingSdkVersion=5.2.1
+RNIap_playBillingSdkVersion=5.1.0
 
 android.useAndroidX=true
 android.enableJetifier=true
```
Then try to build your project :'D

**Comment by MuhammadRafeh** [comment]
@ahmedrowaihi Brother I simply updated the kotlin version to "1.8.0" in project/build.gradle and it worked :)

**Comment by MuhammadRafeh** [comment]
ext {
        buildToolsVersion = "33.0.0"
        minSdkVersion = 21
        compileSdkVersion = 33
        targetSdkVersion = 33
        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
        ndkVersion = "23.1.7779620"
        kotlinVersion = "1.8.0"
        androidXAnnotation = "1.1.0"
        androidXBrowser = "1.0.0"
    }

**Comment by ahmedrowaihi** [comment]
Nice, project's React Native version is 0.69, and using expo modules with version 46..
So updating my kotlin version hits everywhere conflicting with others libs :'(

**Comment by coofzilla** [comment]
same issue, i'm on:

```kotlin
    ext {
        buildToolsVersion = "35.0.0"
        minSdkVersion = 24
        compileSdkVersion = 35
        targetSdkVersion = 35
        ndkVersion = "27.1.12297006"
        kotlinVersion = "2.0.21"
    }
```

and 

``` "react-native": "~0.79.0",```

**edit**

Upgrading to ```"react-native-iap": "~12.16.0",``` resolved it for me.

**Comment by andyken28** [comment]
ext {
        buildToolsVersion = "36.0.0"
        minSdkVersion = 24
        compileSdkVersion = 36
        targetSdkVersion = 36
        ndkVersion = "27.1.12297006"
        kotlinVersion = "2.1.20"
    }
And
"dependencies": {
    "@react-native/new-app-screen": "0.81.0",
    "react": "19.1.0",
    "react-native": "0.81.0",
    "react-native-iap": "^13.0.4",
    "react-native-safe-area-context": "^5.5.2"
  },
error :react-native-iap:compilePlayDebugKotlin FAILED
please help me ~~~~

**Comment by TomaszNiedzielski** [comment]
@andyken28 I have the same config and same error. Have you fix it?

**Comment by ganesht10** [comment]
+1

**Comment by hyochan** [comment]
#2986 has been released! I’ll go ahead and close this for now. Please feel free to open a new issue if the problem persists.

**Comment by danishsshaikh** [comment]
thank you @hyochan, for 14.0.0 - the issue of `react-native-iap:compilePlayDebugKotlin FAILED` had been resolved with the nitro modules <3

**Comment by Sameer22-724** [comment]
**### Notice**
Starting from version 14.0.0, this library uses [Nitro Modules](https://github.com/mrousavy/nitro) for high-performance native bridge implementation. You must **install react-native-nitro-modules alongside react-native-iap**.
